packages/extension-ui/src/Popup/Signing/Signing.test.tsx

```diff
@@ -0,0 +1,194 @@
+    });
+
+    it('only ArrowLeft should be active on second screen', () => {
+      wrapper.find('ArrowRight').simulate('click');
+      expect(wrapper.find('ArrowLeft').prop('isActive')).toBe(true);
+      expect(wrapper.find('ArrowRight').prop('isActive')).toBe(false);
+      expect(wrapper.find(TransactionIndex).text()).toBe('Transaction:2/2');
+    });
+
+    it('should display previous request after ArrowLeft has been clicked', () => {
+      wrapper.find('ArrowRight').simulate('click');
+      wrapper.find('ArrowLeft').simulate('click');
+      expect(wrapper.find(TransactionIndex).text()).toBe('Transaction:1/2');
+      expect(wrapper.find(Request).prop('signId')).toBe(signRequests[0].id);
+    });
+  });
+
+  describe('Request rendering', () => {
+    it('correctly displays request 1', () => {
+      expect(wrapper.find(Address).text()).toBe('acc15D4bqjQRPgdMBK8bNvhX4tSuCtSGZS7rZjD5XH5SoKcFeKn5');
+      expect(wrapper.find(Details).find('td.data').map((el): string => el.text())).toEqual([
+        'polkadot.js',
+        'Alexander',
+        '112',
+        '0',
+        'balances.transfer',
+        `{
+  "dest": "5D4bqjQRPgdMBK8bNvhX4tSuCtSGZS7rZjD5XH5SoKcFeKn5",
+  "value": 31415926500000
+}`,
+        ' Transfer some liquid free balance to another account.   `transfer` will set the `FreeBalance` of the sender and receiver.  It will decrease the total issuance of the system by the `TransferFee`.  If the sender\'s account is below the existential deposit as a result  of the transfer, the account will be reaped.   The dispatch origin for this call must be `Signed` by the transactor.',
+        'mortal, valid from #4,033,610 to #4,033,674'
+      ]);
+    });
+
+    it('correctly displays request 2', () => {
+      wrapper.find('ArrowRight').simulate('click');
+      expect(wrapper.find(Address).text()).toBe('acc25E9nq1yGJJFiP8C75ryD9J2R62q2cesz6NumLnuXRgmuN5DG');
+      expect(wrapper.find(Details).find('td.data').map((el): string => el.text())).toEqual([
+        'polkadot.js',
+        'Alexander',
+        '112',
+        '0',
+        'balances.transfer',
+        `{
+  "dest": "5D1ss3KFnzNtLzRDfUhqLivzVvt5BDrBnK21dMf1si2twPuj",
+  "value": 10000000000
+}`,
+        ' Transfer some liquid free balance to another account.   `transfer` will set the `FreeBalance` of the sender and receiver.  It will decrease the total issuance of the system by the `TransferFee`.  If the sender\'s account is below the existential deposit as a result  of the transfer, the account will be reaped.   The dispatch origin for this call must be `Signed` by the transactor.',
+        'mortal, valid from #4,034,205 to #4,034,269'
+      ]);
+    });
+  });
+
+  describe('Submitting', () => {
+    it('passes request id to cancel call', () => {
+      wrapper.find('CancelButton').find('a').simulate('click');
+      expect(messaging.cancelSignRequest).toBeCalledWith(signRequests[0].id);
```

Comment:
I would argue this test can end at this line. We have already tested that clicking arrow right changes the underlying `signRequest`.